hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/conf/Configuration.java

```diff
@@ -3718,52 +3709,42 @@ public static void dumpConfiguration(Configuration config,
    */
   public static void dumpConfiguration(Configuration config,
       Writer out) throws IOException {
-    JsonFactory dumpFactory = new JsonFactory();
-    JsonGenerator dumpGenerator = dumpFactory.createGenerator(out);
-    dumpGenerator.writeStartObject();
-    dumpGenerator.writeFieldName("properties");
-    dumpGenerator.writeStartArray();
-    dumpGenerator.flush();
+    JsonWriter jsonWriter = new JsonWriter(out);
+    jsonWriter.beginObject().name("properties").beginArray().flush();
```

Comment:
do we have tests for this writing of configs? If not, they're needed first

Reply:
We have, but as you said below, if we purge jackson completely, there is no way to ensure that the json format is not changed, as in tests we also use gson...